### PR TITLE
Maintenance: Remove change-minimizing MemObject::Io hack

### PR DESCRIPTION
**Pull request by vshailesh**
Added in 2018 commit 4310f8b0.

### CODE REVIEW

File: src/MemObject.h [hunk 1]
```diff
@@ -168,19 +168,17 @@ class MemObject
...
-    /* TODO: Remove this change-minimizing hack */
-    using Io = Store::IoStatus;
-    static constexpr Io ioUndecided = Store::ioUndecided;
-    static constexpr Io ioReading = Store::ioReading;
-    static constexpr Io ioWriting = Store::ioWriting;
-    static constexpr Io ioDone = Store::ioDone;
+    static constexpr Store::IoStatus ioUndecided = Store::ioUndecided;
+    static constexpr Store::IoStatus ioReading = Store::ioReading;
+    static constexpr Store::IoStatus ioWriting = Store::ioWriting;
+    static constexpr Store::IoStatus ioDone = Store::ioDone;
```
Comment (rousskov) [inline src/MemObject.h:174]: >  TODO: Remove this change-minimizing hack
   
Here, "this hack" includes `Io` declaration _and_ four static members that use that `Io` declaration. Please remove _all_ of them and adjust the affected code to use Store::IoStatus directly, avoiding any shims.

```suggestion
```

**Review by rousskov — CHANGES_REQUESTED** [review]

**Comment by rousskov** [comment]
I have adjusted PR title to be more specific and use `Maintenance:` prefix that tells us that this change does not affect Squid behavior in any way (and, hence, is safe to backport, skip during bug-searching bisection, etc.). @vshailesh, please keep this change in mind when naming future code maintenance PRs.

I also adjusted PR description to supply useful information instead of narrating arguably self-documenting diff.

**Comment by vshailesh** [comment]
@rousskov I have made the changes in all the files that were using `MemObject::io****`. 

Please review the changes

File: src/MemObject.h (hunk 1 above)
Comment (vshailesh) [reply src/MemObject.h]: removed these 4 lines and replaced with Store::IoStatus wherever necessary

**Pull request closed without merging**